Pokedex/ListViewController.swift

```diff
@@ -0,0 +1,340 @@
+    
+    func setupCollectionView(){
+        let layout = UICollectionViewFlowLayout()
+        layout.minimumLineSpacing =  4
+        layout.minimumInteritemSpacing = 2
+        let frame = CGRect(x: 0, y: sc.frame.height + UIApplication.shared.statusBarFrame.height +
+            self.navigationController!.navigationBar.frame.height + 1, width: view.frame.width, height: view.frame.height)
+        collectionView = UICollectionView(frame: frame, collectionViewLayout: layout)
+        collectionView.contentInset = UIEdgeInsetsMake(0, 0, self.bottomLayoutGuide.length + 120, 0)  //this is to prevent tab bar from hiding last cell
+        collectionView.register(PokemonCollectionViewCell.self, forCellWithReuseIdentifier: "pokemonCell")
+        collectionView.backgroundColor = UIColor.white
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        view.addSubview(collectionView)
+        
+    }
+    //Create a method with a completion handler to get the image data from your url
+    
+//
+//    func getDataFromUrl(url: URL, completion: @escaping (_ data: Data?, _  response: URLResponse?, _ error: Error?) -> Void) {
+//        URLSession.shared.dataTask(with: url) {
+//            (data, response, error) in
+//            completion(data, response, error)
+//            }.resume()
+//    }
+    //Create a method to download the image (start the task)
+    
+//    func downloadImage(url: URL) {
+//        print("Download Started")
+//        getDataFromUrl(url: url) { (data, response, error)  in
+//            DispatchQueue.main.sync() { () -> Void in
+//                guard let data = data, error == nil else { return }
+//                print(response?.suggestedFilename ?? url.lastPathComponent)
+//                print("Download Finished")
+//                self.pokemonImageView.image = UIImage(data: data)
+//            }
+//        }
+//    }
+
+    
+    //handles changes in Segmented Control Value
+//    func changeSegment(sender: UISegmentedControl){
+//        print("Changing Segment!!")
+//        if sender.selectedSegmentIndex == 1{
+//            print("tableview selected")
+//            tableView.isHidden = true
+//        }else{
+//            print("collection selected")
+//            tableView.isHidden = false
+//        }
+//        
+//    }
+    
+}
+
+
+
+extension ListViewController: UITableViewDataSource,UITableViewDelegate{
+    //Specify number of cells in tableview
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 20
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "pokemonCell", for: indexPath) as! PokemonTableViewCell
+        for subview in cell.contentView.subviews{
+            subview.removeFromSuperview()
+        }
+        cell.awakeFromNib()
+        self.tableView?.rowHeight = 80
```

Comment:
consider using dimensions of UIScreen.mainscreen whenever making UI decisions and setting heights or widths of objects of your app so that everything scales based on user's device. So instead of 80 for height, try UIScreen.mainscreen().bounds.height/20.